src/panel/pages/timeline/components/TimelineTooltip.tsx

```diff
@@ -15,24 +15,37 @@ export interface TooltipPosition {
   y: number;
 }
 
-export const TimelineTooltip: FC<JSX.IntrinsicElements["div"]> = ({
-  children,
-  ...props
-}) => (
-  <div {...props} style={{ paddingBottom: 8, zIndex: 9, ...props.style }}>
-    <TooltipElement>{children}</TooltipElement>
-  </div>
-);
-
-const TooltipElement = styled.p<{ position?: TooltipPosition }>`
+type TimelineTooltipProps = {
+  handleTooltipRef?: (passedRef: any) => void;
+  tooltipOffset?: number;
+};
+
+export const TimelineTooltip: FC<
+  JSX.IntrinsicElements["div"] & TimelineTooltipProps
+> = ({ children, handleTooltipRef, tooltipOffset, ...props }) => {
+  const offset = useMemo(() => (tooltipOffset ? tooltipOffset : 0), [
+    tooltipOffset,
+  ]);
+  return (
+    <div
+      {...props}
+      style={{ marginLeft: tooltipOffset, ...props.style }}
+      ref={handleTooltipRef}
+    >
+      <TooltipElement offset={offset}>{children}</TooltipElement>
+    </div>
+  );
+};
+
+const TooltipElement = styled.p<{ position?: TooltipPosition; offset: number }>`
   position: relative;
   background-color: ${(p) => p.theme.dark["-3"]};
   border-radius: 2px;
   color: #fff;
   font-size: 12px;
   margin: 0;
   padding: 10px 20px;
-  z-index: 9;
+  white-space: nowrap;
```

Comment:
Is it possible to allow wrapping with these adjustments? 

My thinking is that longer messages would likely need to be wrapped so a max-width on the tooltip would probably make sense.

Reply:
Yeah this thought did cross my mind. Part of me thought that blocking the wrapping could be a good influence to stop us having tooltips which are too long. I'll take a look to see if there's a way we can be more dynamic with it

Reply:
> Part of me thought that blocking the wrapping could be a good influence to stop us having tooltips which are too long.

That's a fair point! Sounds like there's no technical limitation so we can always change this later if needed 🙏